templates/minimal/runtime/tests/validate_dev_chain_spec.rs

```diff
@@ -0,0 +1,87 @@
+use sc_service::ChainType;
+use sc_chain_spec::{
+    ChainSpecExtension,
+    ChainSpecGroup,
+    Properties,
+};
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use core::str;
+use std::error::Error;
+use std::io::BufReader;
+
+pub type ChainSpec = sc_service::GenericChainSpec<Extensions>;
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize, ChainSpecGroup, ChainSpecExtension)]
+pub struct Extensions {
+    #[serde(alias = "relayChain", alias = "RelayChain")]
+    pub relay_chain: String,
+    #[serde(alias = "paraId", alias = "ParaId")]
+    pub para_id: u32,
+}
+
+impl Extensions {
+    pub fn try_get(chain_spec: &dyn sc_service::ChainSpec) -> Option<&Self> {
+        sc_chain_spec::get_extension(chain_spec.extensions())
+    }
+}
+
+
+fn decode_runtime_info(spec_data: Value) -> 
+    Result<(String, u32, u32), Box<dyn Error>> {
+    let runtime_hx_cde = spec_data
+        .get("genesis").expect("failed to get genesis")
+        .get("runtimeGenesis").expect("failed to get runtimeGenesis")
+        .get("code").expect("failed to get code")
+        .as_str().expect("failed to turn to str");
+
+    let clean_u8 = runtime_hx_cde.trim_start_matches("0x").as_bytes(); // normalize runtime hex cde remove '0x' and convert to bytes.
+    let version_info: RuntimeVersion = RuntimeVersion::decode(&mut &clean_u8[..])
+    .map_err(|_| "Failed to decode runtime version")?;
+
+    let rt_spec_name = version_info.spec_name.into_owned();
+    let rt_spec_version = version_info.spec_version;
+    let rt_authoring_version = version_info.authoring_version;
+
+    Ok((rt_spec_name, rt_spec_version, rt_authoring_version))
+}
+
+#[test]
+fn test_minimal_dev_chain_spec_rt_validity() {
+    let mut properties = Properties::new();
+    properties.insert("tokenDecimals".to_string(), 12.into());
+    properties.insert("tokenSymbol".to_string(), "UNIT".into());
+    let current_wasm = WASM_BINARY.expect("Development wasm not available");
+
+    let test_chain_spec: serde_json::Value = serde_json::from_str(ChainSpec::builder(
+        current_wasm,
+        Extensions {
+            relay_chain: "dev".into(),
+            para_id: 1000,
+        })
+        .with_name("test_Development")
+        .with_id("dev")
+        .with_chain_type(ChainType::Local)
+        .with_genesis_config_preset_name(sp_genesis_builder::DEV_RUNTIME_PRESET)
+        .with_properties(properties.clone())
+        .build()
+        .as_json(false)
+        .unwrap()
+        .as_str()
+    ).unwrap();
+
+    let existing_chain_spec_file =
+		std::fs::File::open("../dev_chain_spec.json").expect("file should open. qed");
+	let existing_chain_spec_reader = BufReader::new(existing_chain_spec_file);
+	let existing_chain_spec: serde_json::Value =
+		serde_json::from_reader(existing_chain_spec_reader).expect("should read proper JSON. qed");
+
+    let data1 = decode_runtime_info(existing_chain_spec).unwrap();
+    let data2 = decode_runtime_info(test_chain_spec).unwrap(); 
```

Comment:
My intention was to avoid these, and assert directly on the JSONs equivalence. Does the test fail on your end if not adding these, when comparing the JSONs? It might be because of the difference that I mentioned. The test needs the chain_specs to be generated based on a runtime built with `cargo build`, without the `--release`profile.

Reply:
My bad  🤦🏾. I am going to go and do it with out the --release

Reply:
Hey @iulianbarbu  please correct me where I am wrong.
My understanding of the chain specification, particularly the fields, started from this link: 

https://paritytech.github.io/polkadot-sdk/master/sc_chain_spec/index.html#genesis-initial-runtime-state. 

Specifically, I looked at the genesis, which includes the Code or the top JSON representation of the chain specification found at https://paritytech.github.io/polkadot-sdk/master/sc_chain_spec/index.html. 

Since the test chain specification is created with the name `with_name("test_Development")`, simply comparing the JSONs may not yield clear results. Therefore, I traversed the JSON structure to locate the code value, which I can then use for comparison.

Therefore, I use `sp_version::RuntimeVersion` and pass it the code value to access runtime API, specifically the `spec_name`, `spec_version`, and `authoring_version` as referenced in https://paritytech.github.io/polkadot-sdk/master/sp_version/struct.RuntimeVersion.html.

Reply:
> Does the test fail on your end if not adding these, when comparing the JSONs?

Yes it does fail

Reply:
> Since the test chain specification is created with the name with_name("test_Development"), simply comparing the JSONs may not yield clear results.

You are right. Sorry for the confusion, I missed changing this when suggested the change. The idea is to modify `with_name`, `with_id` & `chain_type` to be the same as those inside the `dev_chain_spec.json`. You can open the `dev_chain_spec.json` and see the values for `name`, `id`, and `chain_type`. And then, combined with the idea of generating the `dev_chain_spec.json` with `cargo build` instead of `cargo build --release`, the created chain spec in the test converted to a  `serde_json::Value` should be the same with the `serde_json::Value` read from `dev_chain_spec.json`.

Reply:
I see that test approaching  🟢
Making a push with those specifications

Reply:
Hey @iulianbarbu 
I added the test, and it passed successfully. Everything is working as expected.
Let me proceed to create the one for the parachain I guess unless there are other changes to look at